pkg/mcs/tso/server/server.go

```diff
@@ -316,7 +323,74 @@ func (s *Server) initClient() error {
 	return err
 }
 
-func (s *Server) startServer() (err error) {
+func (s *Server) startGRPCServer(l net.Listener) {
+	defer s.serverLoopWg.Done()
+
+	gs := grpc.NewServer()
+	s.service.RegisterGRPCService(gs)
+	err := gs.Serve(l)
+	if err != cmux.ErrListenerClosed {
+		log.Fatal("grpc server stops serving unexpectedly.", errs.ZapError(err))
+	}
+	log.Info("gRPC server stop serving.", errs.ZapError(err))
+
+	// Attempt graceful stop (waits for pending RPCs), but force a stop if
+	// it doesn't happen in a reasonable amount of time.
+	done := make(chan struct{})
+	go func() {
+		gs.GracefulStop()
+		close(done)
+	}()
+	select {
+	case <-done:
+	case <-time.After(defaultGRPCGracefulStopTimeout):
+		log.Info(fmt.Sprintf("Stopping grpc gracefully is taking longer than %v."+
+			" Force stopping now. Pending RPCs will be abandoned.", defaultGRPCGracefulStopTimeout))
+		gs.Stop()
+	}
+}
+
+func (s *Server) startHTTPServer(l net.Listener) {
+	defer s.serverLoopWg.Done()
+
+	handler, _ := SetUpRestHandler(s.service)
+	hs := &http.Server{
+		Handler:           handler,
+		ReadTimeout:       5 * time.Minute,
+		ReadHeaderTimeout: 5 * time.Second,
+	}
+	err := hs.Serve(l)
+	if err != cmux.ErrListenerClosed {
+		log.Fatal("http server stops serving unexpectedly.", errs.ZapError(err))
+	}
+	log.Info("http server stop serving. Taking more http(s) requests.", errs.ZapError(err))
```

Comment:
take more?

Reply:
changed to log.Info("http server stop serving. processing accepted requests and shutdowning the server now.", errs.ZapError(err))